great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1329,3 +1324,38 @@ def get_batch_data_and_markers(
             )
 
         return batch_data, batch_markers
+
+    def __del__(self):
+        """Close any open connections on shutdown."""
+        if self._connection:
+            self._connection.close()
+
+    # TODO: What is return type, docstring
+    def execute_query(self, query: Selectable):
+        """Execute a query using the underlying database engine.
+
+        Some databases sqlite/mssql temp tables only persist within a connection,
+        so we need to keep the connection alive.
```

Comment:
@anthonyburdi
I think this is fine.
But just to throw out some more ideas... you could still do this as a context manager function.
And conditionally close the connection on exit if it's not `GXSqlDialect.SQLITE` or `GXSqlDialect.MSSQL`

Or you could move that control to the caller so it's more explicit and just make this `excecute_query(query: Selectable, close_connection_on_exit: bool)`

Reply:
@Kilo59 I gave this a try in the most recent commit. I needed to keep the `__del__` to make sure that the connection is closed when the instance of the engine is removed, but in the test I need to call this via `execution_engine.__del__()` and not via `del execution_engine` which is surprising.

Open to any suggestions here.

Reply:
I'm also considering trying to use a [StaticPool](https://docs.sqlalchemy.org/en/20/core/pooling.html#sqlalchemy.pool.StaticPool) instead for databases that need to use the same connection each time.